### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by vyatsykiv**
Bloc architecture on Profile screen.

### CODE REVIEW

File: lib/blocs/profile/profile_state.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+abstract class ProfileState {
+
+}
+
+class LoadingState extends ProfileState {
+
+}
+
+class DisplayProfileState extends ProfileState {
```
Comment (ababicheva) [inline lib/blocs/profile/profile_state.dart]: let's rename it to ProfileLoadedState. right now it's a bit confusing for me

File: lib/blocs/profile/profile_block.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import 'package:flutterapp/blocs/profile/profile_state.dart';
+import 'package:flutterapp/data/data_load_result.dart';
+import 'package:flutterapp/data/model/user.dart';
+import 'package:flutterapp/data/repository/user_repository.dart';
+
+class ProfileBloc extends Bloc<ProfileEvent, ProfileState> {
+
+  final UserRepositoryAPI userRepository;
+
+  bool _isExpanded = false;
```
Comment (ababicheva) [inline lib/blocs/profile/profile_block.dart]: I'm not sure about isExpanded state. should we have shared states (like user) and ephemeral states (like isExpanded) in blocs? I would rather pass isExpanded as a param with ExpandAchievementsEvent as that's a widget's state

File: lib/data/network_manager.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:connectivity/connectivity.dart';
+
+///abstract class to mock it during tests
+abstract class NetworkManagerAPI {
```
Comment (ababicheva) [inline lib/data/network_manager.dart]: what does -API postfix mean here?
this one is responsible for connectivity checking, right?
let's name it smth like ConnectivityManager or ConnectivityChecker and rename it's implementation correspondingly.

File: lib/data/repository/user_repository.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import 'package:flutterapp/data/model/achievement.dart';
+import 'package:flutterapp/data/model/user.dart';
+import 'package:flutterapp/data/network_manager.dart';
+import 'package:flutterapp/resources/app_images.dart';
+
+abstract class UserRepositoryAPI {
+  Future<DataLoadResult<User>> getUserProfile();
+}
+
+class UserRepository implements UserRepositoryAPI {
```
Comment (ababicheva) [inline lib/data/repository/user_repository.dart]: should we implement api or have an instance of it in the repo? let's discuss

File: lib/data/network_manager.dart (hunk 3 above)
Comment (vyatsykiv) [reply lib/data/network_manager.dart]: It's just another way how to name interface to describe what we have to implement. I don't like INetworkManager cause of it's even more ugly than NetworkManagerAPI. 
'Rename it's implementation correspondingly' - what exactly to you mean? 
MobileConnectivityManager and MockConnectivityManager?

File: lib/blocs/profile/profile_block.dart (hunk 2 above)
Comment (vyatsykiv) [reply lib/blocs/profile/profile_block.dart]: When user click on Expand button Bloc must to know current widget state to know what state send to widget.

Comment (ababicheva) [reply lib/blocs/profile/profile_block.dart]: we can pass isExpanded with ExpandAchievementsEvent

File: lib/data/network_manager.dart (hunk 3 above)
Comment (ababicheva) [reply lib/data/network_manager.dart]: I just don't see any reason to add API postfix here

File: lib/blocs/profile/profile_state.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutterapp/data/model/user.dart';
+
+abstract class ProfileState extends Equatable {
```
Comment (yakivmospan) [inline lib/blocs/profile/profile_state.dart]: Not really sure if we need `Equatable`? Is It common to use this in Dart?

Comment (yakivmospan) [reply lib/blocs/profile/profile_state.dart]: And how it is behaving, when there is no params in object?

File: lib/blocs/profile/profile_block.dart (hunk 2 above)
Comment (yakivmospan) [reply lib/blocs/profile/profile_block.dart]: Let's just rename this to something for Bloc related, rather view related. Like isAchivementsVisible or so

File: test/mocks/network_manager_mock.dart [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import 'package:flutterapp/data/network_manager.dart';
+import 'package:mockito/mockito.dart';
+
+NetworkManager getConnectionManager(bool hasConnection) {
```
Comment (yakivmospan) [inline test/mocks/network_manager_mock.dart:4]: Are there any libraries that allow us to create mocks as Mockito does?

File: lib/blocs/profile/profile_block.dart (hunk 2 above)
Comment (vyatsykiv) [reply lib/blocs/profile/profile_block.dart]: Renamed to _isAchievementsExpanded as achievements are always visible (3 achievements are visible by default).

File: lib/blocs/profile/profile_state.dart (hunk 5 above)
Comment (vyatsykiv) [reply lib/blocs/profile/profile_state.dart]: Can say that it's common but you can find this framework in a lot of articles.
Regarding what if no parameters? In such case equatable take runtimeType and identical(). //if two objects are the same.

But in current version there is no more needs in equatable. Removing it.

File: test/mocks/network_manager_mock.dart (hunk 6 above)
Comment (vyatsykiv) [reply test/mocks/network_manager_mock.dart:4]: You can use Mockito on Flutter as well.

**Review by ababicheva — APPROVED** [review]

File: pubspec.yaml [hunk 7]
```diff
@@ -28,6 +28,7 @@ dependencies:
   flutter_bloc: ^4.0.0
   connectivity: ^0.4.8+6
   equatable: ^1.2.0
+  flutter_simple_dependency_injection: any
```
Comment (yakivmospan) [inline pubspec.yaml]: That's a bad practice, please set a concrete version instead of `any` and we are good to go with merging.

**Review by yakivmospan — APPROVED** [review]

**Pull request merged**